iceoryx2-bb/lock-free/src/spmc/unrestricted_atomic.rs

```diff
@@ -73,26 +74,142 @@ impl<T: Copy> Producer<'_, T> {
         // After writing the content of the write_cell, the content needs to be synced with the
         // reader.
         /////////////////////////
-        self.atomic.write_cell.fetch_add(1, Ordering::Release);
+        self.atomic.mgmt.write_cell.fetch_add(1, Ordering::Release);
     }
 }
 
 impl<T: Copy> Drop for Producer<'_, T> {
     fn drop(&mut self) {
-        self.atomic.has_producer.store(true, Ordering::Relaxed);
+        self.atomic.mgmt.has_producer.store(true, Ordering::Relaxed);
     }
 }
 
 unsafe impl<T: Copy> Send for Producer<'_, T> {}
 unsafe impl<T: Copy> Sync for Producer<'_, T> {}
 
+#[doc(hidden)]
+#[repr(C)]
+pub struct UnrestrictedAtomicMgmt {
+    write_cell: IoxAtomicU32,
+    has_producer: IoxAtomicBool,
+}
+
+impl Default for UnrestrictedAtomicMgmt {
+    fn default() -> Self {
+        Self {
+            write_cell: IoxAtomicU32::new(1),
+            has_producer: IoxAtomicBool::new(true),
+        }
+    }
+}
+
+impl UnrestrictedAtomicMgmt {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// # Safety
```

Comment:
Please keep the safety section but still add the attribute `#[doc(hidden)]` to all `__internal` functions.